### DISCUSSION TITLE
ReadRelationships: what is the difference between `relationship_filter.optional_relation` and nested `subject_filter.optional_relation`?

### DISCUSSION THREAD

**Discussion by theronic** [authzed/spicedb#1861] [Q&A] [answered]
Refer [ReadRelationships filters](https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.RelationshipFilter).

**Comment by vroldanbet** [comment]
Hi @theronic, If I understand your question correctly, `ReadRelationshipRequest.optional_relation` is not a thing, it's `optional_limit`

**Comment by theronic** [comment]
@vroldanbet see ReadRelationshipRequest.relationship_filter: https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.RelationshipFilter

Then look inside `optional_subject_filter`.

**Reply by vroldanbet** [reply, marked as answer]
@theronic in SpiceDB the subject optional relation allows you to reference, instead of a subject entity, a relation in a subject entity. If you think of a graph, what this allows you to is to reference an edge instead of a node. Example:

````
definition user {}
 
definition team {
  relation member: user
}
 
definition resource {
  relation reader: user | team#member
  permission view = reader
}
```

In this example, the resource `reader` relation can be assigned to a user or to the members of a team. The `member` part of `team#member` is denoted with `optional_subject_filter.optional_relation`

**Reply by theronic** [reply]
@vroldanbet isn't filtering on the `member` relation covered by `relationship_filter.optional_relation = "member"`?

**Reply by vroldanbet** [reply]
@theronic no, that's the relation on the resource, not on the subject.

```
resource:myres#reader@team:engineering#members  
``` 
- reader is the resource relation
- members is the subject relation

**Reply by theronic** [reply]
Thanks. In the example above, would #members be #member with the schema definition?

**Reply by vroldanbet** [reply]
@theronic sorry, I meant to write `member`. Yes, it's `team#memter` from the schema

**Comment by theronic** [comment]
Updated title to clarify where these filters are.
